### PR TITLE
Allows ai's to eject from mech domination

### PR DESCRIPTION
**Pull request by Rex9001**
## About The Pull Request
Lets ais eject from dominated mechs

If this is too strong at 30 I'd be happy to bump it or apply some other restrictions
## Why It's Good For The Game
They have a big eject button which is essentially bait at the moment, it does nothing. 

I think its fine to let combat upgraded ai's and malfunctional ai's to use this mech domination ability more freely, because how it usually goes is malf ai dominates mech, everyone sees occupants of mech get tossed out and the ai is promptly killed. I don't think this is fun or healthy for an ability which is pretty expensive and the only other way I see it used is an ai just trying to live for a few moments more by shunting into a cargo mech, from which they can't do a whole lot and simply exist until the crew figure them out.

TLDR: For a very expensive ability it does not provide a lot of benefit and is easily dispatched by the crew. 
## Changelog
:cl:
balance: Ai's using the mech domination ability can now eject from the mech
/:cl:

### CODE REVIEW

**Comment by Rex9001** [comment]
TGUI check my beloved

**Comment by xzero314** [comment]
This might also fix a weird bug that currently exists for combat upgraded ais. Basically if you are a combat upgraded ai and you enter a mech the normal way that all ais can (without even unlocking mech dominate) the game treats it as if you dominated the mech and you need an admin to get you back out

**Comment by ExcessiveUseOfCobblestone** [comment]
So what does the eject button do now?

Im guessing this lets them hopscotch from say the core to mech to nearby apc (prev just core to mech to gg)?

**Comment by Rex9001** [comment]
> So what does the eject button do now?
> 
> Im guessing this lets them hopscotch from say the core to mech to nearby apc (prev just core to mech to gg)?

Eject button ejects the ai to their core. The apc thing suffers from similar issues in that how this functions currently is a you buy it you lose. I'm working on a bunch of other prs currently and if this is the desired effect of core->mech->apc I could implement it in a future pr.

**Comment by ExcessiveUseOfCobblestone** [comment]
i dont have a pref but ideally there wouldnt be so much hopscotch (mech to apc is fine) that it becomes a nuisance, i dont think jumping jellybean ai is fun either.

**Comment by Rex9001** [comment]
Personally I think its fine for the price, and jumping jellybean malf ai seems more in flavor of the antagonist that is exploiting technology all across the station rather than just controlling one mech and then dying forever.

**Comment by ExcessiveUseOfCobblestone** [comment]
sorry i keep giggling about jumping jellybean ai and the imagery it evokes
